src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -607,6 +632,27 @@ private void ExecutePutSubscriptionBatch(TransactionOperationContext context, Bl
                 {
                     ExecuteManyOnDispose(context, index, type, action.Value);
                 }
+
+                long GetMinimumSubscriptionIdFor(string database)
+                {
+                    if (minimumSubscriptionIdPerDatabase.TryGetValue(database, out var minimumSubscriptionId))
+                        return minimumSubscriptionId;
+
+                    minimumSubscriptionId = int.MaxValue;
+
+                    foreach (var keyValue in ReadValuesStartingWith(context, SubscriptionState.SubscriptionPrefix(database)))
```

Comment:
This does a scan on all subscriptions on put, no? Is there any way to avoid that?
I understand that we are now doing this once on put batch instead on each subscription, but that is still not ideal

Reply:
I can save the minimum value and update it after the command is completed.